src/util/metrics/mod.rs

```diff
@@ -60,14 +60,15 @@ pub fn run_prometheus(
     Some(handler)
 }
 
-pub fn dump() -> String {
+pub fn dump() -> Result<String, Error> {
     let mut buffer = vec![];
     let encoder = TextEncoder::new();
     let metric_familys = prometheus::gather();
     for mf in metric_familys {
         if let Err(e) = encoder.encode(&[mf], &mut buffer) {
-            warn!("ignore prometheus encoding error: {:?}", e);
+            warn!("prometheus encoding error: {:?}", e);
+            return Err(e);
```

Comment:
I think we should continue encoding other metrics when there are errors.

Reply:
But how to let HTTP server know there is an error?

Reply:
Maybe it doesn't need to know? Also seems that HTTP server may not have many action options for the error. For metrics with encoding errors, it won't be shown in the result, which will finally result in missing metrics in the dashboard.

Reply:
I think we don't need to return the error here. 
Just skipping is fine.